RELEASE_CHECKLIST.md

```diff
@@ -424,6 +425,91 @@ Start following the steps below to start a release PR.  The process is **not ful
     curl -s "https://github.com/apollographql/router/releases/tag/v${APOLLO_ROUTER_RELEASE_VERSION}" | htmlq 'meta[property="og:image"]' --attribute content
     ```
 
+### Reconciling `main` back to `dev`
```

Comment:
This is better documented than it was before, though I should make a callout to this section in the Finalizing step.  I also applied one learning that never made it into the previous version of this doc, which was how to actually reliably fix merge conflicts in the event that you needed to.